src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double CAPACITY_MULTIPLIER = 1.5;
+    private static final int DEFAULT_CAPACITY = 10;
+
+    private T[] elementsData;
+    private int size;
+
+    public ArrayList() {
+        this.elementsData = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
+    private void indexExistence(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounds: " + index);
+        }
+    }
+
+    private void resizeCheck(int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index for add operation: "
+                    + index);
+        }
+    }
+
+    private void ensureCapacity(int minCapacity) {
```

Comment:
Are you creating a new variable name just for one method?
And it's unnecessary, you can call class-level variables.

Reply:
Not sure, kinda burn with it few times(